src/toast/__tests__/toaster.test.js

```diff
@@ -0,0 +1,361 @@
+      expect(
+        wrapper
+          .find(Toaster)
+          .first()
+          .props().usePortal,
+      ).toBe(true);
+      expect(ReactDOM.createPortal).toHaveBeenCalled();
+    });
+
+    test('portal is not created when usePortal is set to false', () => {
+      wrapper = mount(toaster.create({usePortal: false}));
+      expect(
+        wrapper
+          .find(Toaster)
+          .first()
+          .props().usePortal,
+      ).toBe(false);
+      expect(ReactDOM.createPortal).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('toaster overrides', () => {
+    test('Inner components override', () => {
+      const overrides = {
+        Root: jest
+          .fn()
+          .mockImplementation(({children}) => <div>{children}</div>),
+        ToastBody: jest
+          .fn()
+          .mockImplementation(({children}) => <div>{children}</div>),
+        ToastCloseIcon: jest
+          .fn()
+          .mockImplementation(({children}) => <svg>{children}</svg>),
+      };
+      // $FlowFixMe
+      wrapper = mount(toaster.create({overrides}));
+      const renderedToaster = wrapper.find(Toaster).first();
+      toaster.getRef = jest.fn().mockReturnValue(renderedToaster.instance());
+      toaster.show('Toast message');
+      wrapper.update();
+
+      Object.keys(overrides).forEach(name => {
+        expect(wrapper.find(overrides[name]).first()).toExist();
+      });
+    });
+
+    test('Inner components props and styles overrides', () => {
+      const overrides = {
+        Root: {
+          props: {'data-attr': 'root'},
+          style: {borderColor: 'skyblue'},
+        },
+        ToastBody: {
+          props: {'data-attr': 'body'},
+          style: {backgroundColor: 'skyblue'},
+        },
+        ToastCloseIcon: {
+          props: {'data-attr': 'closeIcon'},
+          style: {color: 'blue'},
+        },
+      };
+      const components = {
+        Root: StyledRoot,
+        ToastBody: StyledBody,
+        ToastCloseIcon: StyledCloseIcon,
+      };
+      wrapper = mount(toaster.create({overrides}));
+
+      const renderedToaster = wrapper.find(Toaster).first();
+      toaster.getRef = jest.fn().mockReturnValue(renderedToaster.instance());
+      toaster.show('Toast message');
+      wrapper.update();
+
+      Object.keys(overrides).forEach(name => {
+        const passedProps = wrapper
+          .find(components[name])
+          .first()
+          .props();
+        expect(passedProps['data-attr']).toEqual(
+          overrides[name].props['data-attr'],
+        );
+        expect(passedProps.$style).toEqual(overrides[name].style);
+      });
+    });
+  });
+
+  describe('toaster methods', () => {
+    let renderedToaster;
+
+    beforeEach(() => {
+      wrapper = mount(toaster.create());
+      renderedToaster = wrapper.find(Toaster).first();
+      toaster.getRef = jest.fn().mockReturnValue(renderedToaster.instance());
+    });
+
+    test('toaster[show | update | clear]', () => {
```

Comment:
do you think we can split this into multiple test cases? it seems to be super long

Reply:
Only the show part could be separated, but the update/clear still need some toasts to be shown first so it would be included anyway.